numpy/lib/npyio.py

```diff
@@ -784,6 +783,162 @@ def floatconv(x):
     else:
         return asstr
 
+
+def skip(skipcols):
+    r"""
+    Helper function for the `usecols` parameter of `loadtxt` and `genfromtxt`.
+
+    The parameter `usecols` of `loadtxt` and `genfromtxt` allows the user to
+    select specific columns from the text file.  This `skip` function provides
+    a simple way for the user to specify the columns that should be *skipped*
+    instead of the columns that should be used.
+
+    Parameters
+    ----------
+    skipcols : int or sequence of ints
+        The indices of the columns to be skipped.
+
+    Returns
+    -------
+    f : function
+        Returns a function with the signature::
+
+            def f(n: int) -> List[int]
+
+        that may be passed to the functions `numpy.loadtxt` and
+        `numpy.genfromtxt`.  The function is a closure that "rememebers"
+        the list of columns that were passed to `skip`.  The function
+        returns ``list(range(n))`` with the values from `skipcols` removed.
+
+    Examples
+    --------
+    >>> from numpy.lib.npyio import skip
+    >>> s = " 0  1  2  3  4  5\n10 11 12 13 14 15\n20 21 22 23 24 25"
+    >>> print(s)
+     0  1  2  3  4  5
+    10 11 12 13 14 15
+    20 21 22 23 24 25
+
+    Use `skip` to skip the second column (column index 1) and the last
+    column (column index -1).
+
+    >>> np.loadtxt(s.splitlines(), dtype=int, usecols=skip([1, -1]))
+    array([[ 0,  2,  3,  4],
+           [10, 12, 13, 14],
+           [20, 22, 23, 24]])
+
+    To see what `skip` actually does, call the returned function with
+    ``n = 6``:
+
+    >>> skip([1, -1])(6)
+    [0, 2, 3, 4]
+    """
+    try:
+        skipcols = list(skipcols)
+    except TypeError:
+        # Presumably skipcols is an int, so wrap it in a list.
+        skipcols = [skipcols]
```

Comment:
```suggestion
    skipcols = _normalize_usecols_to_list(skipcols)
```

Reply:
I gave a thumbs up reaction, but I realized this won't quite work, because `_normalize_use_cols_to_list` requires `n`, the number of fields, and we don't have that when `skip` is called.

Reply:
But I like the idea that `skip` accepts the same things that `usecols` does, and using it results in the complement of the set of the fields selected by its argument, whatever the format of the argument.  That means `skip` doesn't return a function, but is an object that `loadtxt` must be aware of.   I'll think about it.

Reply:
It can return the function, as long as it defers the normalization until it's called.